napari/_qt/qt_event_loop.py

```diff
@@ -238,6 +237,19 @@ def gui_qt(*, startup_logo=False, gui_exceptions=False, force=False):
     IPython with the Qt GUI event loop enabled by default by using
     ``ipython --gui=qt``.
     """
+    warn(
+        "\nThe 'gui_qt()' context manager is deprecated.\nIf you are running "
```

Comment:
Should we say "and will be removed in `0.5.0`"? Seems appropriate to leave this around until then

Reply:
I think @tlambert03 left that out because I suggested in the past that we should have a longer deprecation cycle for this, since so many people will be using this API... If we want to set a release on it I would suggest 0.6. What do you think @sofroniewn?

Reply:
`0.6.0` is fine with me, though it feels a long way out! I don't see that maintaining this API is causing us problems right now, so ok to have a long drop time. @tlambert03 you ok with this?

Reply:
other than _really_ wanting people not to use it, I don't have any specific objection :).  I'd almost rather leave it unspecified (so we can go more by the number of months and not some ill defined future point).

Reply:
yeah, the uncertainty around what and when `0.6.0` would even be isn't ideal for me, but i think specifying could help people feel less taken by surprise when it does go. i'll defer to @jni final opinion here

Reply:
I think we can defer the precise timing of the removal to a future PR. When it does come time to remove it, we can have a final two-patch-release notice. In the meantime I'm just going to merge this as-is! 🎉

Reply:
Thank you for your service `gui_qt()`! ❤️